src/OpenDB/src/lefout/lefout.cpp

```diff
@@ -42,15 +43,20 @@
 
 using namespace odb;
 
-void lefout::writeBoxes(void* boxes, const char* indent)
+void lefout::writeVersion(const char* version)
+{
+  fprintf(_out, "VERSION %s ;\n", version);
+}
+
+void lefout::writeBoxes(dbSet<dbBox>* boxes, const char* indent)
 {
-  dbSet<dbBox>* geoms = (dbSet<dbBox>*) boxes;
-  dbSet<dbBox>::iterator bitr;
   dbTechLayer* cur_layer = NULL;
 
-  for (bitr = geoms->begin(); bitr != geoms->end(); ++bitr) {
-    dbBox* box = *bitr;
+  for (dbBox* box : *boxes) {
     dbTechLayer* layer = box->getTechLayer();
+    if (box == nullptr) {
+      fprintf(_out, "THIS is basically null!\n");
+    }
```

Comment:
wouldn't this have segfaulted on the previous line? Ie. this is probably not needed.